presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -101,6 +104,10 @@
     protected final String connectionUrl;
     protected final Properties connectionProperties;
     protected final String identifierQuote;
+    private final boolean mapTableNames;
+
+    private Map<String, Map<String, String>> schemaTableMapping = new HashMap<>();
```

Comment:
> I can extract that to a separate Class. Is it preferred to be an inner class for this type of usage? Or a properly separate class file?

Up to you. Choose the thing what fits bets.

Regarding the tests, sorry for not reading the commit message. What about using PSQL instead of H2?